python/pyspark/sql/utils.py

```diff
@@ -24,7 +28,11 @@ def __init__(self, desc, stackTrace):
         self.stackTrace = stackTrace
 
     def __str__(self):
-        return repr(self.desc)
+        desc = self.desc
+        if isinstance(desc, unicode):
+            return str(desc.encode('utf-8'))
```

Comment:
cc @zero323 and @davies too. Would you have some time to take a look for this one? This is a typical annoying problem between unicode and byte strings. There are many similar PRs (at least I can identify few PRs trying to handle this problem. One good example might help resolving other PRs too.